HIP/hip-1340.md

```diff
@@ -0,0 +1,247 @@
+
+```
+message CryptoCreateTransactionBody {
+...
+/**
+ * The delegation indicator of authorized contract address for the account.
+ * It will be encoded as defined in EIP-7702 as (0xef0100 || <contract address>).
+ * If this field is set, a call the account's address within a smart contract will
+ * result in the code of the authorized contract being executed.
+ */
+ bytes delegation_indicator=19
+}
+
+and 
+
+message CryptoUpdateTransactionBody {
+...
+/**
+ * The delegation indicator of authorized contract address for the account.
+ * It will be encoded as defined in EIP-7702 as (0xef0100 || <contract address>).
+ * If this field is set, a call the account's address within a smart contract will
+ * result in the code of the authorized contract being executed.
+ */
+ bytes delegation_indicator=19
+}
+```
+
+In addition to these protobuf updates, functionality will be added in the `TokenService` to call the 
+`ContractService` to save the association between the `account` and the `authorized_contract_delegation_indicator` 
+into `ContractStateStore`.  This will be accomplished via the creation of a new `ContractServiceApi` whose 
+interface will be defined as follows:
+
+```
+/**
+ * Defines mutations that can't be expressed as a {@link com.hedera.hapi.node.transaction.TransactionBody} dispatch.
+ *
+ * <p>Only exported to the token service at this time, as it is the only service that currently needs such a thing.
+ */
+public interface ContractServiceApi {
+    /**
+     * Sets the account identified by the accountID with the associated runtime bytecode in the ContractStateStore.
+     *
+     * @param accountID id of the account to set the bytecode for
+     * @param bytecode  the runtime bytecode to set for the account
+     */
+    void setAccountBytecode(@NonNull AccountID accountID, @NonNull Bytes bytecode);
+}
+```
+
+### Impact on Mirror Node and Block Node
+
+Mirror Node will be impacted in the following ways:
+
+- EOAs will store a mutable code
+- The code (as returned by ContractGetBytecode) of HTS tokens and HSS schedule addresses will be updated (from custom EVM snippets to EIP-7702 delegation indicators)
+- The code of EOA accounts (as returned by ContractGetBytecode) will be updated (from custom EVM snippets to either empty or delegation indicator)
+
+### Impact on RPC Relay
+
+- RPC Relay will need to support type 4 transactions
+- RPC Relay will need to update its internal/hardcoded bytecode of HTS and HSS tokens
+
+### Impact on SDK
+
+All supported SDKs will need to support parsing, creating and submitting type 4 transactions.
+
+## Backwards Compatibility
+
+The change itself is backwards compatible, but it significantly modifies the behavior of calls to EOAs and invalidates some previous assumptions (e.g. the code of an EOA always being empty).
+
```

Comment:
I suggest to a add a top level Protocol Changes Summary section. where we include

- Protobuf changes
- AccountUpdate and AccountCreate changes (cryptoUpdate, CryptoCreate). Direct or synthetic execution

- Record and BlockStream implications
- Type 4 transaction , parsing
- Proxy precedence
- Security v2 update o reflect 7702